controllers/databasecluster_controller.go

```diff
@@ -133,7 +134,8 @@ func (r *DatabaseClusterReconciler) reconcileDB(
 		if done, err := p.Cleanup(ctx, db); err != nil {
 			return ctrl.Result{}, err
 		} else if !done {
-			return ctrl.Result{Requeue: true}, nil
+			db.Status.Status = everestv1alpha1.AppStateDeleting
+			return ctrl.Result{Requeue: true}, r.Status().Update(ctx, db)
```

Comment:
I think we should set the status regardless of the value of `done`. What happens if `Cleanup` returns `done=true` the first time it is called, but the foregroundDeletion is still pending. In that case, we may never see a `deleting` state on the DBC, even though it is actually being deleted.

```diff
diff --git a/controllers/databasecluster_controller.go b/controllers/databasecluster_controller.go
index fb0313b..9f56419 100644
--- a/controllers/databasecluster_controller.go
+++ b/controllers/databasecluster_controller.go
@@ -130,12 +130,14 @@ func (r *DatabaseClusterReconciler) reconcileDB(
 ) (ctrl.Result, error) {
        // Handle any necessary cleanup.
        if !db.GetDeletionTimestamp().IsZero() {
-               if done, err := p.Cleanup(ctx, db); err != nil {
+               done, err := p.Cleanup(ctx, db)
+               if err != nil {
                        return ctrl.Result{}, err
-               } else if !done {
-                       return ctrl.Result{Requeue: true}, nil
                }
-               return ctrl.Result{}, nil
+               db.Status.Status = everestv1alpha1.AppStateDeleting
+               return ctrl.Result{
+                       Requeue: !done,
+               }, r.Status().Update(ctx, db)
        }

        // Set metadata.
```

Reply:
Right, thanks, updated👍